docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,349 @@
 
-## How to start Async Code in F# #
+This is a key difference with the C#/VB style of `async` programming. In F#, asynchronous computations can be thought of as **Cold tasks**. They must be explicitly started to actually execute. This has some advantages, as it allows you to combine and sequence asynchronous work easily.
 
-As mentioned earlier, async code is a specification of work to be done in another context which needs to be explicitly started. Here are two primary ways to accomplish this:
+## Combining asynchronous computations
 
-1.  `Async.RunSynchronously` will start an async workflow on another thread and await its result.
+Here is an example that builds upon the previous one by combining computations:
 
 ```fsharp
 open System
-open System.Net
+open System.IO
 
-let fetchHtmlAsync url = 
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
-        let! html = webClient.AsyncDownloadString(uri)
-        return html
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
     }
 
- // Execution will pause until fetchHtmlAsync finishes
- let html = "https://dotnetfoundation.org" |> fetchHtmlAsync |> Async.RunSynchronously
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Parallel
+    |> Async.Ignore
+    |> Async.Start
 
- // you actually have the result from fetchHtmlAsync now!
- printfn "%s" html
- ```
+    Console.Read() |> ignore
+    0 // return an integer exit code
+```
+
+As you can see, the `main` function has quite a few more calls made. Conceptually, it does the following:
+
+1. Transform the command-line arguments into `Async<unit>` computations with `Array.map`
+2. Create an `Async<'T[]>` that schedules and runs the `countFileBytes` computations in parallel when it runs.
+3. Create an `Async<unit>` that will run the parallel computation and ignore its result
+4. Explicitly run the last computation with `Async.Start`
 
-2.  `Async.Start` will start an async workflow on another thread, and will **not** await its result.
+When this program runs, `countFileBytes` runs in parallel for each command-line argument. Because asynchronous computations execute independently of program flow, there is no order in which they print their information and finish executing.
+
+## Sequencing asynchronous computations
+
+Because `Async<'T>` is a specification of work rather than a computation that is currently running, you can perform more intricate transformations easily. For example, here is an implementation of the Sequence monad that converts a list of `Async<'T>` into an `Async<'T>` that contains an array of results. It uses the previous `countFileBytes` function:
```

Comment:
specification of work is a good term
"computation  that is currently running" is problematic for me - suggest "running task"

>you can perform more intricate transformations easily.

you can use the fact that you have a set of specifications / builders to build up a quite different overall computation from those same building blocks thanks to the fact that the actual scheduling is always deferred ?

>into an `Async<'T>` that contains an array of results

into an `Async<'T[]>` that will yield the aggregated results from the individual tasks at such a point as the full set becomes available.